popup/Calendar/CreateCalendar.md

```diff
@@ -0,0 +1,284 @@
+
+**Return value**: Text (HTML)
+
+![image](https://github.com/user-attachments/assets/c6b00255-c5d5-4f51-9c50-659ba12d8382)
+
+
+
+**Example** (run on final line as a function) :
+
+Returns 
+
+
+
+```js
+var event_day =[{"Name":"Trash","Color":"#849434","Events":["2025-04-03T00:00:00-07:00","2025-04-10T00:00:00-07:00","2025-04-17T00:00:00-07:00","2025-04-24T00:00:00-07:00","2025-05-01T00:00:00-07:00","2025-05-08T00:00:00-07:00","2025-05-15T00:00:00-07:00","2025-05-22T00:00:00-07:00","2025-05-29T00:00:00-07:00","2025-06-05T00:00:00-07:00","2025-06-12T00:00:00-07:00","2025-06-19T00:00:00-07:00"]},{"Name":"Recycle","Color":"#5592B4","Events":["2025-04-07T00:00:00-07:00","2025-04-13T00:00:00-07:00","2025-04-20T00:00:00-07:00","2025-04-27T00:00:00-07:00","2025-05-04T00:00:00-07:00","2025-05-11T00:00:00-07:00","2025-05-18T00:00:00-07:00","2025-05-25T00:00:00-07:00","2025-06-01T00:00:00-07:00","2025-06-08T00:00:00-07:00","2025-06-15T00:00:00-07:00","2025-06-22T00:00:00-07:00"]},{"Name":"Green Waste","Color":"#B47755","Events":["2025-04-07T00:00:00-07:00","2025-05-09T00:00:00-07:00"]}]
+
+ CreateCalendar(event_day,True,"The next scheduled {variable} pick up is ","MMMM D, Y",2,"#1a6324","#FFFFFF","#8c8c8c","49%","250px")
+```
+
+## Expression
+
+```js 
+// Define the attributes for the event. The Name, color to show on the calendar and the dates of the event(s). 
+var event_day = []
+
+/*Change display to show next event or calendar*/
+var show_calendar = True
+
+/* The text to use before the next event if show_calendar is false. Use "{variable}"" to insert the event name
+ex. "The next scheduled Event is " */
+var display_text ="The next scheduled {variable} pick up is "
+
+/*The date format used when the calendar is not displayed. See https://developers.arcgis.com/arcade/function-reference/text_functions/#text  */
+var date_format = "MMMM D, Y"
+
+/*Specify the number of months to display. Only applicable if show_calendar is true*/
+var num_months = 2
+
+/*Indicate color of header on calendar. Only applicable if show_calendar is true*/
+var header_color="#1a6324"
+
+/*Indicate color of the text on the calendar's header. Only applicable if show_calendar is true*/
+var header_text_color = "#FFFFFF"
+
+/*Indicate color of "today" outline on calendar. Only applicable if show_calendar is true*/
+var today_color = "#8c8c8c"
+
+/*Indicate default max width of each calendar*/
+var max_width = "49%"
+
+/*Indicate the minimum width of each calendar */
+var min_width = "250px"
+
+
+
+/*DO NOT CHANGE ANYTHING BELOW THIS LINE
+-------------------------------------------------------------------------*/
+
+function CreateCalendar(events,showCalendar,displayText,dateFormat,numberMonths,headerColor,headerTextColor,todayColor,minWidth,maxWidth){
+ 	//Check inputs and set defaults
+	if(IsEmpty(events)){
+		return("Event List Required")
+	}
+
+	iif(IsEmpty(showCalendar),showCalendar=True,showCalendar)
```

Comment:
You can remove all these and just use DefaultValue later, would be a lot cleaner